kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/SolverServiceBase.java

```diff
@@ -480,31 +480,28 @@ private void solvePlanningProblem(final SolverInstanceContext sic,
                                       final Object planningSolution) {
         sic.getInstance().setBestSolution(null);
         sic.getInstance().setStatus(SolverInstance.SolverStatus.SOLVING);
-        this.executor.execute(
-                new Runnable() {
-                    @Override
-                    public void run() {
-                        try {
-                            // If the executor's queue is full, it's possible that the solver gets canceled before it starts
-                            SolverInstance.SolverStatus status;
-                            synchronized (sic) {
-                                status = sic.getInstance().getStatus();
-                                // TODO Race condition: status turns into non-solving before solver starts
-                                // See https://issues.jboss.org/browse/PLANNER-540
-                            }
-                            if (status == SolverInstance.SolverStatus.SOLVING) {
-                                sic.getSolver().solve(planningSolution);
-                            }
-                        } catch (Exception e) {
-                            logger.error("Exception executing solver '" + sic.getInstance().getSolverId() + "' from container '" + sic.getInstance().getContainerId() + "'. Thread will terminate.",
-                                         e);
-                        } finally {
-                            synchronized (sic) {
-                                sic.getInstance().setStatus(SolverInstance.SolverStatus.NOT_SOLVING);
-                            }
-                        }
-                    }
-                });
+        this.executor.execute(() -> {
+            try {
+                // If the executor's queue is full, it's possible that the solver gets canceled before it starts
+                SolverInstance.SolverStatus status;
+                synchronized (sic) {
+                    status = sic.getInstance().getStatus();
+                    // TODO Race condition: status turns into non-solving before solver starts
+                    // See https://issues.jboss.org/browse/PLANNER-540
+                }
+                if (status == SolverInstance.SolverStatus.SOLVING) {
+                    sic.getSolver().solve(planningSolution);
+                }
+            } catch (Exception e) {
+                logger.error("Exception executing solver '" + sic.getInstance().getSolverId()
+                                     + "' from container '" + sic.getInstance().getContainerId() + "'. Thread will terminate.",
+                             e);
```

Comment:
Can you open a jira that we need to report that error back to the user when he comes asking for the result in a new REST request?